samcli/local/layers/layer_downloader.py

```diff
@@ -92,6 +92,13 @@ def download(self, layer, force=False):
         if layer.is_defined_within_template:
             LOG.info("%s is a local Layer in the template", layer.name)
             layer.codeuri = resolve_code_path(self.cwd, layer.codeuri)
+
+            if layer.codeuri.lower().endswith('.zip'):
```

Comment:
We should follow the same thing we do for functions here. Can you abstract [this method out](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/lambdafn/runtime.py#L152) to be usable here and in the runtime?